### ISSUE TITLE
JSON Schema doesn't allow `headers` for `schema`

### ISSUE THREAD

**Issue by benkroeger** [kamilkisiela/graphql-config#765]
**Describe the bug**
According to the [documentation](https://graphql-config.com/schema#passing-headers), you can pass headers to a schema url - and the feature itself actually works.
It's just that the provided json schema throws a validation error because it expects schema to either be a string or an array of strings

**To Reproduce**
set schema to

```yaml
schema:
  - http://localhost:3000/graphql:
      headers:
        Authorization: foo
```

yaml schema validation against `https://unpkg.com/graphql-config@4.0.1/config-schema.json` will report the issue as follows:

![image](https://user-images.githubusercontent.com/7782055/134351021-daa3e486-33d9-4b86-9c67-d2c084738e7d.png)

**Expected behavior**
the documented format should be allowed in the json schema

**Environment:**

- OS: MacOS 11.5.2
- GraphQL Config Version: 4.0.1
- NodeJS: 16.9.1

**Additional context**

**Comment by paleite** [comment]
Experiencing the same issue

**Comment by B2o5T** [comment]
@benkroeger @paleite @zifeo @kevinalh @illarionvk @BasixKOR @frehner @franzwilhelm

Hey guys, sorry for a long time in fixing this problem.
This problem was finally solved in the alpha version `graphql-config@4.3.0-alpha-21f95f2.0`.

I mentioned all of you because you had this problem, so you can test it on your configs, and maybe you have some suggestions for the next major of `graphql-config`? Please leave it in https://github.com/kamilkisiela/graphql-config/issues/626

**Comment by BasixKOR** [comment]
Note: You can test this by using [custom schema options](https://github.com/redhat-developer/vscode-yaml#associating-a-schema-to-a-glob-pattern-via-yamlschemas) on VS Code YAML extension like this:

```json
"yaml.schemas": {
    "https://unpkg.com/graphql-config@4.3.0-alpha-b9eff39.0/config-schema.json": "graphql.config.*"
}
```
